packages/cli/src/commands/generate/cell/cell.js

```diff
@@ -64,28 +64,24 @@ export const files = async ({
     // override operationName so that its find_operationName
   }
 
-  let operationName = options.query == null ? '' : options.query.trim()
-  if (operationName == '') {
+  let operationName = options.query
+  if (operationName) {
+    const userSpecifiedOperationNameIsUnique = await operationNameIsUnique(
+      operationName
+    )
+    if (!userSpecifiedOperationNameIsUnique) {
+      if (!options.force) {
```

Comment:
I agree using duplicate operation names are bad. I just didn't want to limit the user. If you think it's best I'm happy to go with a blanket deny to duplicate names.

Reply:
> I just didn't want to limit the user. If you think it's best I'm happy to go with a blanket deny to duplicate names.

That's the beauty of being an opinionated framework - we can impose those limits in our tools 🙂

Reply:
Okay just a slightly different mindset to what I'm used to but happy regardless.

Reply:
It's a fine line between giving user full control and encouraging good behaviour. In this case, I think we'd save most people (and teams) a lot of headache by preventing this - because you often won't know what the problem is until much later, perhaps when writing a completely unrelated feature. 

Thanks for humoring me!

Reply:
Thanks for the guidance! I hope all the comments have been addressed.